scripts/utils/kakarot.py

```diff
@@ -287,32 +289,63 @@ async def eth_send_transaction(
 ):
     """Execute the data at the EVM contract to on Kakarot."""
     evm_account = caller_eoa or await get_eoa()
+    nonce = await evm_account.get_nonce()
+
+    payload = {
+        "type": 0x2,
+        "chainId": KAKAROT_CHAIN_ID,
+        "nonce": nonce,
+        "gas": gas,
+        "maxPriorityFeePerGas": int(1e19),
+        "maxFeePerGas": int(1e19),
+        "to": to_checksum_address(to) if to else None,
+        "value": value,
+        "data": data,
+    }
+
+    typed_transaction = DynamicFeeTransaction.from_dict(payload)
 
-    typed_transaction = TypedTransaction.from_dict(
-        {
-            "type": 0x2,
-            "chainId": KAKAROT_CHAIN_ID,
-            "nonce": await evm_account.get_nonce(),
-            "gas": gas,
-            "maxPriorityFeePerGas": int(1e19),
-            "maxFeePerGas": int(1e19),
-            "to": to_checksum_address(to) if to else None,
-            "value": value,
-            "data": data,
-        }
-    )
     evm_tx = EvmAccount.sign_transaction(
         typed_transaction.as_dict(),
```

Comment:
you could just put the payload here, eventually we don't **need** this Transaction object

Reply:
actually doing TypedTransaction.from_dict will pop the "type" key in the dict, I can also do it manually

Reply:
let's use the object, but then use it 100%